homeassistant/components/idasen_desk/__init__.py

```diff
@@ -116,6 +122,24 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
     entry.async_on_unload(entry.add_update_listener(_async_update_listener))
 
+    @callback
+    def _async_bluetooth_callback(
+        service_info: bluetooth.BluetoothServiceInfoBleak,
+        change: bluetooth.BluetoothChange,
+    ) -> None:
+        """Update from a Bluetooth callback to ensure that a new BLEDevice is fetched."""
+        _LOGGER.debug("Bluetooth callback triggered")
+        hass.async_create_task(coordinator.async_ensure_connection_state())
```

Comment:
It looks like there is nothing preventing multiple of these tasks from running at once

Reply:
Since the library already takes care of that, I think there is no need to make the code more complex on HA side:

https://github.com/abmantis/idasen-ha/blob/main/idasen_ha/connection_manager.py#L43

Reply:
I don't see any locks there.  

What if two connect tasks start at about the same time and the first one yields to the event loop when self._connect is called allowing the second task to start before self._connecting is set?   

This probably won't happen as it likely will never suspend when calling self._connect, but I don't think there is any guarantee that it's safe because self._connecting isn't set before the await that might return control to the event loop.

Reply:
I was under the impression that the call to `_connect` would be immediate, and would only return to the eventloop when it blocks. Is that not the case with asyncio?

Reply:
I have refactored idasen-ha a bit, to try to make it more reliable. Feel free to take a look and let me know if you have any suggestions.